Inject migrate action regardless of allocate action

joegallo opened the pull request:
As part of https://github.com/elastic/elasticsearch/issues/76147, we're no longer going to skip injecting a `MigrateAction`s based on properties an associated `AllocateAction` in the same phase. There are still other circumstances when we would not inject a `MigrateAction` (e.g. when the user has already defined one), though.

Commit by commit is best, IMHO, but there's not a ton of diff here.

joegallo committed:
Remove this extraneous MigrateAction

joegallo committed:
Combine these tests

IMHO the details don't matter much, and randomness captures the
"doesn't matter / either way" components just fine.

joegallo committed:
Reorganize these just a little bit

It reads more clearly to me this way

joegallo committed:
Ignore the allocate action when injecting migrate

joegallo committed:
Get these tests passing

joegallo committed:
One last round of test cleanup

joegallo committed:
Oops, typo

dakrone reviewed (approved):
LGTM

EDIT: needs documentation changes however

joegallo committed:
Merge branch 'master' into inject-migrate-action-regardless-of-allocate

joegallo committed:
Update the ilm-migrate docs

Automatic migration is no longer conditional on whether or not there's
an applicable allocate action.

joegallo committed:
Better updates, and fixes a few missed spots

joegallo committed:
Just one more spot
